File: JavaScript/Ejercicio8.js
function printArray(arr) {
    return `[${arr.join(', ')}]`;
}

function partition(arr, l, h) {
    let pvt = arr[h];
    let j = l - 1;

    for (let k = l; k < h; k++) {
        if (arr[k] < pvt) {
            j += 1;
            swap(arr, j, k);
        }
    }

    swap(arr, j + 1, h);
    return j + 1;
}

function swap(arr, j, k) {
    [arr[j], arr[k]] = [arr[k], arr[j]];
}

function quickSort(arr, l, h) {
    if (l < h) {
        let pi = partition(arr, l, h);

        quickSort(arr, l, pi - 1);
        quickSort(arr, pi + 1, h);
    }
}

function main() {
    const arr = [54, 24, 15, 2, 12, 1, 80];

    console.log("=== QUICK SORT EN JAVASCRIPT ===");
    console.log(`Array inicial: ${printArray(arr)}`);

    quickSort(arr, 0, arr.length - 1);

    console.log(`Array final: ${printArray(arr)}`);
}

main();

File: C#/Ejercicio8.cs
using System;

void QuickSort(int[] arr, int l, int h)
{
    if (l < h) {
        int pi = Partition(arr, l, h);

        QuickSort(arr, l, pi - 1);
        QuickSort(arr, pi + 1, h);
    }
}

int Partition(int[] arr, int l, int h)
{
    int pvt = arr[h];
    int j = l - 1;

    for (int k = l; k < h; k++) {
        if (arr[k] < pvt) {
            j += 1;
            Swap(arr, j, k);
        }
    }

    Swap(arr, j + 1, h);
    return j + 1;
}

void Swap(int[] arr, int j, int k)
{
    int temp = arr[j];
    arr[j] = arr[k];
    arr[k] = temp;
}

int[] arr = {64, 34, 25, 12, 22, 11, 90};
Console.WriteLine("=== QUICK SORT EN C# SCRIPT ===");
Console.WriteLine("Array inicial: [" + string.Join(", ", arr) + "]");

QuickSort(arr, 0, arr.Length - 1);

Console.WriteLine("Array final: [" + string.Join(", ", arr) + "]");